src/GUI/ComboBoxColorPicker.java

```diff
@@ -1,42 +1,43 @@
+package GUI;
+
 import javax.swing.*;
 import java.awt.*;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 
 class ComboBoxColorPicker extends JPanel {
     private final Color[] graphColors = {Color.RED, Color.BLUE, Color.YELLOW, Color.GREEN, Color.GRAY};
-    private final JComboBox<Color> colorPicker = new JComboBox<>(graphColors);   // Combo Box to pick the Color of the Function
+    private final JComboBox<Color> colorPicker = new JComboBox<>(graphColors);
+    public Color currentColor;
 
     public ComboBoxColorPicker() {
-
         setLayout(new BorderLayout());
         colorPicker.setRenderer(new ColorListRenderer());
         colorPicker.addActionListener(new ColorBoxListener());
+        currentColor = (Color) colorPicker.getSelectedItem();
         add(colorPicker, BorderLayout.CENTER);
     }
 
     private class ColorBoxListener implements ActionListener {
         @Override
         public void actionPerformed(ActionEvent e) {
-            Color selectedColor = (Color) colorPicker.getSelectedItem(); // Gets the selected Color from the ComboBox (cast to Color)
-            //TODO: Connect to the color setter of the function
+            currentColor = (Color) colorPicker.getSelectedItem();
```

Comment:
Without currentColor, this whole actionListener can be omitted.

Reply:
Done